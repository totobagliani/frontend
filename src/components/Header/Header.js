import React from 'react';
import { Link } from 'react-router-dom';
import { HomeOutlined, FileAddOutlined } from '@ant-design/icons';
import styles from './styles.module.scss';
import SearchBar from '../SearchBar/SearchBar';

export default function Header() {
  return (
    <div className={styles['header-container']}>
      <header className={styles['App-header']}>
        <h1 className={styles.title}>Products Searcher</h1>
        <nav className={styles['App-header__nav']}>
          <ul className={styles['App-header__list']}>
            <li className={styles['App-header__item']}>
              <Link to="/" className={styles['App-header__link']}>
                <p className={styles['App-header__text']}>HomePage</p>
                <HomeOutlined className={styles['App-header__icon']} />
              </Link>
            </li>
            <li className={styles['App-header__item']}>
              <Link to="/addproduct" className={styles['App-header__link']}>
                <p className={styles['App-header__text']}>Add Product</p>
                <FileAddOutlined className={styles['App-header__icon']} />
              </Link>
            </li>
            <li className={styles['App-header__item']}>
              <SearchBar />
            </li>
          </ul>
        </nav>
      </header>
    </div>
  );
}
