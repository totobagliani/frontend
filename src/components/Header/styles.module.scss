.App-header__icon {
  color: red;
}
