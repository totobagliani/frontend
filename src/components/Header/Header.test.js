/* eslint-disable comma-dangle */
/* eslint-disable linebreak-style */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from '../../redux/store';
import Header from './Header';

const initialState = {
  products: {
    products: [],
    productAdded: {
      productName: '',
      description: '',
      imageURL: '',
      isFavourite: false,
      price: 0,
      section: ''
    },
    searchResults: {
      itemsCount: 0,
      searchTerm: ''
    }
  },
  ui: {
    currentPage: 1,
    currentFilter: '',
    currentOrder: '',
    resultsVisible: false
  }
};
const store = configureStore(initialState);

describe('Given the Header Component', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );
  });
  test('then must be rendered (take a snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment(<Header />)).toMatchSnapshot();
  });
});
