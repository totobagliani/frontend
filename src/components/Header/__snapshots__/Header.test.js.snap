// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Header Component then must be rendered (take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="header-container"
  >
    <header
      class="App-header"
    >
      <h1
        class="title"
      >
        Products Searcher
      </h1>
      <nav
        class="App-header__nav"
      >
        <ul
          class="App-header__container"
        >
          <li
            class="App-header__item"
          >
            <a
              class="App-header__link"
              href="/"
            >
              <img
                alt="flat101"
                class="App-header__logo"
                src="flat101.jpg"
              />
            </a>
          </li>
          <li
            class="App-header__item"
          >
            <div
              class="search__container"
            >
              <form>
                <div>
                  <button
                    class="search__button"
                    type="submit"
                  >
                    <div>
                       
                      <text
                        class="search__buttonText"
                      >
                         Busca un Producto =
                      </text>
                      <span
                        aria-label="search"
                        class="anticon anticon-search search__icon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="search"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                          />
                        </svg>
                      </span>
                    </div>
                  </button>
                </div>
              </form>
            </div>
          </li>
          <li
            class="App-header__item"
          >
            <a
              class="App-header__link"
              href="/addproduct"
            >
              <p
                class="App-header__text"
              >
                AÃ±ade un Producto
              </p>
              <span
                aria-label="file-add"
                class="anticon anticon-file-add App-header__icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="file-add"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494zM544 472c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v108H372c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h108v108c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V644h108c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V472z"
                  />
                </svg>
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>
</DocumentFragment>
`;
